recipes/gstreamer/all/conanfile.py

```diff
@@ -0,0 +1,133 @@
+from conans import ConanFile, tools, Meson, VisualStudioBuildEnvironment
+import glob
+import os
+import shutil
+
+required_conan_version = ">=1.29"
+
+class GStreamerConan(ConanFile):
+    name = "gstreamer"
+    description = "GStreamer is a development framework for creating applications like media players, video editors, streaming media broadcasters and so on"
+    topics = ("conan", "gstreamer", "multimedia", "video", "audio", "broadcasting", "framework", "media")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://gstreamer.freedesktop.org/"
+    license = "GPL-2.0-only"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
+        self.requires("glib/2.66.2")
+    generators = "pkg_config"
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def build_requirements(self):
+        self.build_requires("meson/0.54.2")
+        self.build_requires("pkgconf/1.7.3")
+        if self.settings.os == 'Windows':
+            self.build_requires("winflexbison/2.5.22")
+        else:
+            self.build_requires("bison/3.5.3")
+            self.build_requires("flex/2.6.4")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("%s-%s" % (self.name, self.version), self._source_subfolder)
+
+    def _configure_meson(self):
+        meson = Meson(self)
+        defs = dict()
```

Comment:
I think you can use `meson.options`, just like you use `cmake.definitions`.

Reply:
are you talking about an undocumented feature ? I cannot see it in https://docs.conan.io/en/latest/reference/build_helpers/meson.html

Reply:
Looks like it is undocumented.
A quick grep shows that it's only used in recipes, originally by me.
(fribidi/inih/pkgconf)